src/App.js

```diff
@@ -1,5 +1,172 @@
+
+class CarRacing {
+  #cars;
+  #carsMap;
+  #numberOfTry;
+  #racingResult = '';
+
+  constructor(cars, numberOfTry) {
+    this.#cars = cars;
+    this.#numberOfTry = numberOfTry;
+    this.#carsMap = new Map(cars.map((name) => [name, '']));
+  }
+
+  race() {
+    this.#executeRacing();
+    this.#printResult();
+    this.#announceWinner();
+  }
+
+  #executeRacing() {
+    for (let i = 0; i < this.#numberOfTry; i++) {
+      this.#processOneRound();
+    }
+  }
+
+  #processOneRound() {
+    this.#updateRacingStatus();
+    this.#updateRacingResult();
+  }
+
+  #updateRacingStatus() {
+    this.#cars.forEach((name) => {
+      if (this.#shouldMoveForward()) {
+        this.#carsMap.set(
+          name,
+          this.#carsMap.get(name) + '-'
+        );
+      }
+    });
+  }
+
+  #shouldMoveForward() {
+    return MissionUtils.Random.pickNumberInRange(0, 9) >= 4;
+  }
+
+  #updateRacingResult() {
+    this.#carsMap.forEach((value, key) => {
+      this.#racingResult += `${key} : ${value}\n`;
+    });
+    this.#racingResult += '\n';
+  }
+
+  #printResult() {
+    const Console = MissionUtils.Console;
+    Console.print(this.#numberOfTry);
+    Console.print('실행 결과');
+    Console.print(this.#racingResult);
+  }
+
+  #announceWinner() {
+    const fastestDistance = this.#getFastestDistance();
+    const winners = this.#getWinners(fastestDistance);
+    MissionUtils.Console.print(
+      `최종 우승자 : ${winners.toString()}`
+    );
+  }
+
+  #getFastestDistance() {
+    return Math.max(
+      ...[...this.#carsMap.values()].map((v) => v.length)
+    );
+  }
+
+  #getWinners(fastestDistance) {
+    const winners = [];
+    this.#carsMap.forEach((v, k) => {
+      if (v === '-'.repeat(fastestDistance)) {
+        winners.push(k);
+      }
+    });
+    return winners;
+  }
+}
+
+class InputValidator {
+  static validateCarNames(inputNames) {
+    const cars = this.#parseCarNames(inputNames);
+    this.#validateInput(inputNames, cars);
+    return cars;
+  }
+
+  static #parseCarNames(inputNames) {
+    return inputNames
+      .split(',')
+      .filter((name) => name !== '')
+      .map((name) => name.trim());
+  }
+
+  static #validateInput(inputNames, cars) {
+    if (inputNames === '') {
+      throw Error('자동차 이름을 입력해주세요.');
+    }
+    this.#validateCarNamesLength(cars);
+    this.#validateNoDuplicates(cars);
+  }
+
+  static #validateCarNamesLength(cars) {
+    cars.forEach((name) => {
+      if (name.length > 5) {
+        throw Error('자동차 이름은 5자 이하만 가능합니다.');
+      }
+    });
+  }
+
+  static #validateNoDuplicates(cars) {
+    if (cars.length !== new Set(cars).size) {
+      throw Error(
+        '중복되지 않은 자동차 이름을 입력해주세요.'
+      );
+    }
+  }
+
+  static validateNumberOfTry(input) {
+    const numberOfTry = Number(input);
+    this.#validateNumber(numberOfTry);
+    return numberOfTry;
+  }
+
+  static #validateNumber(numberOfTry) {
+    if (!numberOfTry) {
+      throw Error('횟수는 숫자로 입력해주세요.');
+    }
```

Comment:
```suggestion
    if (isNaN(numberOfTry)) {
      throw Error('횟수는 숫자로 입력해주세요.');
    }
```

혹시 이러한 유효성 검증을 진행하고 싶으셨나요??

```js
    if (!numberOfTry) {
      throw Error('횟수는 숫자로 입력해주세요.');
    }
```
단순히 이 메서드 내부만으로 평가한다면, numberOfTry가 falsy한 값일 경우에 대한 분기처리 이기 때문에 undefined, 0, '' 등이 입력으로 들어올때 에러를 던지는 것으로 확인됩니다! 😮